tests/compiler/bindings/esm.ts

```diff
@@ -130,6 +130,12 @@ export function internrefFunction(a: NonPlainObject, b: NonPlainObject): NonPlai
   return a;
 }
 
+export function functionFunction(fn: () => void): () => void {
+  return fn;
+}
```

Comment:
What if call this `fn` inside `functionFunction`? It will work properly?

Reply:
It's essentially the first class function that is passed around by reference, so that should work. JS side doesn't see a JS function, just the internref object.

Reply:
I see